cwms_radar_api/src/main/java/cwms/radar/api/RatingMetadataController.java

```diff
@@ -0,0 +1,155 @@
+    }
+
+    protected DSLContext getDslContext(Context ctx) {
+        return JooqDao.getDslContext(ctx);
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = "office", description = "Specifies the owning office of "
+                            + "the Rating Specs whose data is to be included in the response. If "
+                            + "this field is not specified, matching rating information from all "
+                            + "offices shall be returned."),
+                    @OpenApiParam(name = "rating-id-mask", description = "RegExp that specifies "
+                            + "the rating IDs to be included in the response. If this field is "
+                            + "not specified, all Rating Specs shall be returned."),
+                    @OpenApiParam(name = "page",
+                            description = "This end point can return a lot of data, this "
+                                    + "identifies where in the request you are. This is an opaque"
+                                    + " value, and can be obtained from the 'next-page' value in "
+                                    + "the response."
+                    ),
+                    @OpenApiParam(name = "page-size", type = Integer.class,
+                            description = "How many entries per page returned. "
+                                    + "Default " + DEFAULT_PAGE_SIZE + "."
+                    ),
+            },
+            responses = {
+                    @OpenApiResponse(status = "200",
+                            content = {
+                                    @OpenApiContent(type = Formats.JSONV2, from =
+                                            RatingMetadataList.class)
+                            }
+                    )},
+            tags = {"Ratings"}
+    )
+    @Override
+    public void getAll(Context ctx) {
+        long start = System.nanoTime();
+        String cursor = ctx.queryParamAsClass("page", String.class).getOrDefault("");
+        int pageSize =
+                ctx.queryParamAsClass("page-size", Integer.class).getOrDefault(DEFAULT_PAGE_SIZE);
+
+        String office = ctx.queryParam("office");
+        String ratingIdMask = ctx.queryParam("rating-id-mask");
+
+        String formatHeader = ctx.header(Header.ACCEPT);
+        ContentType contentType = Formats.parseHeaderAndQueryParm(formatHeader, "");
+
+
+        try (final Timer.Context timeContext = markAndTime("getAll");
+             DSLContext dsl = getDslContext(ctx)) {
+            RatingMetadataDao dao = getDao(dsl);
+
+            // I know data
+            RatingMetadataList metadataList = dao.retrieve(cursor, pageSize, office,
+                    ratingIdMask);
+
+            long end = System.nanoTime();
```

Comment:
this might be appropriate to wrap in an additional timer context directory. that way the metrics system can logs the time.

Reply:
Whoops, that log statement was just my curiousity. That wasn't supposed to be there.  On my machine each page takes about 6 seconds.  Not great but its a start.

Reply:
Testing here lacking.  I need to get together with Adam and find out what he's done.